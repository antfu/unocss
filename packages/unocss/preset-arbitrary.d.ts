export * from './dist/preset-arbitrary'
export { default } from './dist/preset-arbitrary'
