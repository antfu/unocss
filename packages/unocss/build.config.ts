import { defineBuildConfig } from 'unbuild'

export default defineBuildConfig([
  {
    name: 'Dual',
    entries: [
      'src/postcss',
    ],
    clean: true,
    declaration: true,
    failOnWarn: false,
    rollup: {
      emitCJS: true,
    },
  },
  {
    name: 'ESM only',
    entries: [
      'src/index',
      'src/webpack',
      'src/vite',
      'src/astro',
      'src/preset-uno',
      'src/preset-icons',
      'src/preset-attributify',
      'src/preset-tagify',
      'src/preset-web-fonts',
      'src/preset-typography',
      'src/preset-wind',
      'src/preset-mini',
    ],
    clean: false,
    declaration: true,
    externals: [
      'vite',
      'astro',
      'webpack',
      '@unocss/webpack',
    ],
  },
])
