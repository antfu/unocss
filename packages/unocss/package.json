{
  "name": "unocss",
  "version": "0.27.0",
  "description": "The instant on-demand Atomic CSS engine.",
  "keywords": [
    "unocss",
    "atomic-css",
    "atomic-css-engine",
    "css",
    "tailwind",
    "windicss"
  ],
  "homepage": "https://github.com/unocss/unocss#readme",
  "bugs": {
    "url": "https://github.com/unocss/unocss/issues"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/unocss/unocss.git"
  },
  "funding": "https://github.com/sponsors/antfu",
  "author": "Anthony Fu <anthonyfu117@hotmail.com>",
  "sideEffects": false,
  "files": [
    "dist",
    "*.d.ts"
  ],
  "exports": {
    ".": {
      "require": "./dist/index.cjs",
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.ts"
    },
    "./preset-attributify": {
      "require": "./dist/preset-attributify.cjs",
      "import": "./dist/preset-attributify.mjs",
      "types": "./dist/preset-attributify.d.ts"
    },
    "./preset-icons": {
      "require": "./dist/preset-icons.cjs",
      "import": "./dist/preset-icons.mjs",
      "types": "./dist/preset-icons.d.ts"
    },
    "./preset-mini": {
      "require": "./dist/preset-mini.cjs",
      "import": "./dist/preset-mini.mjs",
      "types": "./dist/preset-mini.d.ts"
    },
    "./preset-typography": {
      "require": "./dist/preset-typography.cjs",
      "import": "./dist/preset-typography.mjs",
      "types": "./dist/preset-typography.d.ts"
    },
    "./preset-uno": {
      "require": "./dist/preset-uno.cjs",
      "import": "./dist/preset-uno.mjs",
      "types": "./dist/preset-uno.d.ts"
    },
    "./preset-web-fonts": {
      "require": "./dist/preset-web-fonts.cjs",
      "import": "./dist/preset-web-fonts.mjs",
      "types": "./dist/preset-web-fonts.d.ts"
    },
    "./preset-wind": {
      "require": "./dist/preset-wind.cjs",
      "import": "./dist/preset-wind.mjs",
      "types": "./dist/preset-wind.d.ts"
    },
    "./vite": {
      "require": "./dist/vite.cjs",
      "import": "./dist/vite.mjs",
      "types": "./dist/vite.d.ts"
    }
  },
  "main": "dist/index.cjs",
  "module": "dist/index.mjs",
  "types": "index.d.ts",
  "scripts": {
    "build": "unbuild",
    "stub": "unbuild --stub"
  },
  "dependencies": {
    "@unocss/cli": "workspace:*",
    "@unocss/core": "workspace:*",
    "@unocss/preset-attributify": "workspace:*",
    "@unocss/preset-icons": "workspace:*",
    "@unocss/preset-mini": "workspace:*",
    "@unocss/preset-typography": "workspace:*",
    "@unocss/preset-uno": "workspace:*",
    "@unocss/preset-web-fonts": "workspace:*",
    "@unocss/preset-wind": "workspace:*",
    "@unocss/reset": "workspace:*",
    "@unocss/transformer-directives": "workspace:*",
    "@unocss/transformer-variant-group": "workspace:*",
    "@unocss/vite": "workspace:*"
  },
  "engines": {
    "node": ">=14"
  }
}
