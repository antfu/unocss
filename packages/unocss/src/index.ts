import type { UserConfig } from '@unocss/core'

export * from '@unocss/core'
export { default as presetUno } from '@unocss/preset-uno'
export { default as presetAttributify } from '@unocss/preset-attributify'
export { default as presetIcons } from '@unocss/preset-icons'
export { default as presetWebFonts } from '@unocss/preset-web-fonts'
export { default as presetTypography } from '@unocss/preset-typography'
export { default as presetMini } from '@unocss/preset-mini'
export { default as presetWind } from '@unocss/preset-wind'

export function defineConfig<Theme extends {}>(config: UserConfig<Theme>) {
  return config
}
