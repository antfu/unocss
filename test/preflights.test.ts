import { createGenerator } from 'unocss'

describe('preflights', () => {
  test('basic', async() => {
    const uno = createGenerator({
      preflights: [
        {
          getCSS() {
            return '.hello { text: red }'
          },
          layer: 'preflight',
        },
        {
          getCSS() { return undefined },
          layer: 'void',
        },
        {
          getCSS() { return '.default-preflight {}' },
        },
      ],
      presets: [],
    })
    const { css } = await uno.generate('')
    expect(css).toMatchSnapshot()
  })
})
