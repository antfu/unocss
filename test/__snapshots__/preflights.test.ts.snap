// Vitest Snapshot v1

exports[`preflights > basic 1`] = `
"/* layer: preflights */
.default-preflight {}
/* layer: custom */
.hello { text: red }"
`;

exports[`preflights > preflight root can be customized with array 1`] = `
"/* layer: preflights */
.scope-1{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset: ;--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset: ;--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}[data-scope-2]{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset: ;--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset: ;--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}"
`;

exports[`preflights > preflight root can be customized with string 1`] = `
"/* layer: preflights */
:root{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset: ;--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset: ;--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}"
`;
